srml/staking/src/lib.rs

```diff
@@ -16,7 +16,157 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Staking manager: Periodically determines the best set of validators.
+//! # Staking Module
+//! 
+//! <!-- Original author of paragraph: @gavofyork -->
+//! The staking module is the means by which a set of network maintainers (known as "authorities" in some contexts and "validators" in others) are chosen based upon those who voluntarily place funds under deposit. Under deposit, those funds are rewarded under normal operation but are held at pain of "slash" (expropriation) should they be found not to bee discharging their duties properly. 
+//! 
+//! ## Overview 
+//! 
+//! ### Terminology
+//! <!-- Original author of paragraph: @gavofyork -->
+//! 
+//! - Staking: The process of locking up funds for some time, placing them at risk of slashing (loss) in order to become a rewarded maintainer of the network.
+//! - Validating: The process of running a node to actively maintain the network, either by producing blocks or guaranteeing finality of the chain.
+//! - Nominating: The process of placing staked funds behind one or more validators in order to share in any reward, and punishment, they take.
+//! - Stash account: The account holding an owner's funds used for staking.
+//! - Controller account: The account which controls am owner's funds for staking.
+//! - Era: A (whole) number of sessions which is the period that the validator set (and each validator's active nominator set) is recalculated and where rewards are paid out.
+//! - Slash: The punishment of a staker by reducing their funds ([reference](#references)). 
+//! 
+//! ### Goals
+//! <!-- Original author of paragraph: @gavofyork -->
+//! 
+//! The staking system in Substrate NPoS is designed to achieve three goals:
+//! - It should be possible to stake funds that are controlled by a cold wallet.
+//! - It should be possible to withdraw some, or deposit more, funds without interrupting the role of an entity.
+//! - It should be possible to switch between roles (nominator, validator, idle) with minimal overhead.
+//! 
+//! ### Scenarios
+//! 
+//! #### Staking 
+//! 
+//! Almost any interaction with the staking module requires at least one account to become **bonded**, also known as being a **staker**. For this, all that it is needed is a secondary _**stash account**_ which will hold the staked funds. Henceforth, the former account that initiated the interest is called the **controller** and the latter, holding the funds, is named the **stash**. Also, note that this implies that entering the staking process requires an _account pair_, one of which to take the role of the controller and one to be the frozen stash account (any value locked in stash cannot be used, hence called _frozen_). This process in the public API is mostly referred to as _bonding_ via the `bond()` function. 
```

Comment:
"one to take the role of the controller" (remove 'of which')

Reply:
It sounds ok to me but I am quite bad with `formal` English and trust that you have a reason.

Reply:
It doesn't make sense because you have **an** account pair. There are two accounts, but one (singular) pair. To make it explicit, you could say, "one account to take..."